src/Parser/Core/Modules/CancelledCasts.js

```diff
@@ -0,0 +1,79 @@
+import React from 'react';
+
+import Icon from 'common/Icon';
+import { formatMilliseconds, formatNumber, formatPercentage } from 'common/format';
+import Analyzer from 'Parser/Core/Analyzer';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import StatisticBox, { STATISTIC_ORDER } from 'Main/StatisticBox';
+
+const debug = false;
+
+class CancelledCasts extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+  };
+
+  castsCancelled = 0;
+  castsFinished = 0;
+  beginCastSpellId = 0;
+  wasCastStarted;
+
+  static IGNORED_ABILITIES = [
+  ];
+
+  on_byPlayer_begincast(event) {
+    const spellId = event.ability.guid;
+    if (this.constructor.IGNORED_ABILITIES.includes(spellId)) {
+      return;
+    }
+    if (this.wasCastStarted === true) {
+      this.castsCancelled += 1;
+    }
+    this.beginCastSpellId = event.ability.guid;
+    this.wasCastStarted = true;
+  }
+
+  on_byPlayer_cast(event) {
+    const spellId = event.ability.guid;
+    if (this.constructor.IGNORED_ABILITIES.includes(spellId)) {
+      return;
```

Comment:
Technically, each ignored ability is still a finished cast, so you should increment the 'castsFinished' counter before returning.... or you could stop counting all instant casts because you have no choice but to finish them, maybe the more useful stat is the percentage of spells with cast times you finished.

Reply:
Wouldn't incrementing the casts finished for ignored spells just skew the results considering we are ignoring those spells?

I tested this, if you go to events and only look for begin cast. The numbers match the modules total cats

Reply:
Depends on what you mean by "skewed"

Reply:
Currently the module isn't counting instant casts, unless they are spells that have car times that are sometimes instant. Wouldn't incrementing competed/total casts just lower the cancelled percentage?  I'm already not counting instant casts just by excluding things without a begin cast, so I don't see the advantage of counting these instant casts

Reply:
Oh yeah I misread the logic